rest/routing.go

```diff
@@ -368,18 +369,28 @@ func wrapRouter(sc *ServerContext, privs handlerPrivs, router *mux.Router) http.
 			h.logRequestLine()
 
 			// Inject CORS if enabled and requested and not admin port
-			cors := sc.Config.API.CORS
-			if privs != adminPrivs && cors != nil {
-				cors.AddResponseHeaders(rq, response)
-			}
-
 			// What methods would have matched?
 			var options []string
 			for _, method := range []string{"GET", "HEAD", "POST", "PUT", "DELETE"} {
 				if wouldMatch(router, rq, method) {
 					options = append(options, method)
 				}
 			}
+
+			cors := sc.Config.API.CORS
+			firstElement, _, _ := strings.Cut(strings.TrimPrefix(rq.URL.Path, "/"), "/")
+			fmt.Println("firstElement: ", firstElement)
+			dbName, _, _, _ := ParseKeyspace(firstElement)
+			fmt.Println("dbName: ", dbName)
+			if dbName != "" {
+				db, err := h.server.GetActiveDatabase(dbName)
+				if err == nil {
+					cors = db.CORS
+				}
+			}
+			if cors != nil && privs != adminPrivs && privs != metricsPrivs {
+				cors.AddResponseHeaders(rq, response)
+			}
```

Comment:
Proposed changes:
- return the matchInfo from wouldMatch, and use it to check for either {db} or {keyspace} on the matching path
- take the first match (while iterating over methods on line 374)
- use that to swap to db.CORS prior to calling AddResponseHeaders and the remaining code.